src/libraries/System.IO.Ports/src/System/IO/Ports/SerialPort.Win32.cs

```diff
@@ -26,8 +26,10 @@ public static string[] GetPortNames()
                     string[] result = serialKey.GetValueNames();
                     for (int i = 0; i < result.Length; i++)
                     {
-                        // Replace the name in the array with its value.
-                        result[i] = (string)serialKey.GetValue(result[i]);
+                        // Replace the name in the array with its value, trimming at the first null character if it exists
+                        var temp = (string)serialKey.GetValue(result[i]);
+                        var end = temp.IndexOf('\0');
```

Comment:
I am not sure whether we should try to workaround the `null` characters here. Ideally [RegistryKey.GetValueNames](https://github.com/dotnet/runtime/blob/aea45f7aaa8f73cb2a585d8b14ce66634d58cb68/src/libraries/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs#L839) should do it out of the box.

@dotnet/area-microsoft-win32 thoughts?

Reply:
GetValueNames looks like better choice to me. can you please give it try @ctacke ?

Reply:
I'm not sure what the suggestion is.  The Value Name is not the issue, the Value itself is.  I can't "try GetValueNames" as a resolution, because it doesn't give the data I need.

You could certainly argue that `Registry.GetValue` should be stopping at the first null, but I don't know if there are other instances where a null-containing string might be valid data.  If null-containing strings are generally considered invalid data for registry entries, then yes, the bug would be in `Registry.GetValue`

Reply:
I may be wrong but it seems like this is what the method does e.g. terminates on null (while the `GetValue` does not)
https://learn.microsoft.com/en-us/dotnet/api/microsoft.win32.registrykey.getvaluenames?view=net-8.0#microsoft-win32-registrykey-getvaluenames

Perhaps @adamsitnik  can explain more.